### PR TITLE
added route to get transaction data from the plaid api

### PR DESCRIPTION
**Pull request by TomerBenya**

### CODE REVIEW

**Review by sdl433 — COMMENTED** [review]
Looks cool. I'll use some of this code, but I'll be awaiting other ppl's reviews or opinions on package-lock.json though before explicit approval and merging

File: back-end/app.js [hunk 1]
```diff
@@ -79,6 +79,36 @@ app.get("/api/categories", async (req, resp) => {
   }
 });
 
+// function to get transactions from Plaid
+app.get("/api/transactions/get", async (req, resp) => {
+  const request = {
+    access_token: accessToken,
```
Comment (Jennifercheukyin) [inline back-end/app.js:85]: Did you test if the api call works? 
I don't quite understand where you got this accessToken. If you got it from the frontend, you should be parsing `request.body`. If you got it directly in the backend, it's a global constant, `ACCESS_TOKEN`, in roughly line 36.

File: back-end/app.js [hunk 2]
```diff
@@ -79,6 +79,36 @@ app.get("/api/categories", async (req, resp) => {
   }
 });
 
+// function to get transactions from Plaid
```
Comment (Jennifercheukyin) [inline back-end/app.js:82]: Please don't forget to include the Plaid doc here, thank you.

**Review by Jennifercheukyin — CHANGES_REQUESTED** [review]

File: back-end/app.js [hunk 3]
```diff
@@ -79,6 +79,36 @@ app.get("/api/categories", async (req, resp) => {
...
+app.get("/api/transactions/get", async (req, resp) => {
+  const request = {
+    access_token: accessToken,
+    start_date: "2018-01-01",
+    end_date: "2020-02-01",
+  };
+  try {
+    const response = await client.transactionsGet(request);
```
Comment (tangym27) [inline back-end/app.js:90]: what's client here? running this returned a reference error that client was not defined

File: back-end/app.js [hunk 4]
```diff
@@ -79,6 +79,36 @@ app.get("/api/categories", async (req, resp) => {
...
+        end_date: "2020-02-01",
+        options: {
+          offset: transactions.length,
+        },
+      };
+      const paginatedResponse = await client.transactionsGet(paginatedRequest);
+      transactions = transactions.concat(paginatedResponse.data.transactions);
+    }
+  } catch (error) {
+    console.log(error.response.data);
```
Comment (tangym27) [inline back-end/app.js:108]:
```suggestion
    console.log(error);
```

sometimes response can be undefined – happened when I first ran it so I would play it safe and stick with error

**Review by tangym27 — COMMENTED** [review]
I wasn't able to successfully test the route – said that client was not defined? I think you'd probably want to merge with Alan's recent push to master since that saves bank data which then might be the client?

File: back-end/app.js (hunk 2 above)
Comment (TomerBenya) [reply back-end/app.js:82]: Ill fix the problems and add the doc! thanks for the reminder

**Pull request closed without merging**